postgres/manage.sls

```diff
@@ -44,7 +44,25 @@ postgres-reload-modules:
 # Database states
 
 {%- for name, db in postgres.databases|dictsort() %}
-
+  {%- if 'extensions' in db %}
+    {%- for ext_name, extension in db.pop('extensions')|dictsort() %}
+      {%- do extension.update({'maintenance_db': name }) %}
```

Comment:
If we'd look at [CREATE EXTENSION](https://www.postgresql.org/docs/current/static/sql-createextension.html) PG SQL syntax, we see that there is no direct relation between an extension and a database.
So an extension is intended to be installed into a schema, and in turn a schema should be associated with a database.

The `maintenance_db` parameter for all `postges_*.present` state function means PG cluser-wide database containing PostgreSQL internal entities, this is not the database you could install extension to.
So, no need to for the line above (49).

Reply:
Following salt docs for postgresql_extension.present:
maintenance_db
    Database to act on
I've checked and state behave as docs describe. Schema is optional for 'CREATE EXTENSION', default is to create extension for current database.

Reply:
Oh, yes. You're absolutely right. An extension state is little bit different from the others in this case. I have checked the code of `postgres.psql_query` function (which is the workhorse of all those states) and it simply substitutes `maintenance_db` to `psql --dbname {{maintenance_db}}` raw command.
That's quite confusing, because other functions use `dbname` parameter instead of `maintenance_db`, so I misunderstood them. Thanks for pointing me to it.

Reply:
Oh, yes. You're absolutely right. An extension state is little bit different from the others in this case. I have checked the code of `postgres.psql_query` function (which is the workhorse of all those states) and it simply substitutes `maintenance_db` to `psql --dbname {{maintenance_db}}` raw command.
That's quite confusing, because other functions use `dbname` parameter instead of `maintenance_db`, so I misunderstood them. Thanks for pointing me to it.